src/main/java/se/sundsvall/messaging/service/mapper/StatisticsMapper.java

```diff
@@ -81,25 +82,30 @@ public static Statistics toStatistics(final List<StatsEntry> stats) {
             .build();
     }
 
-    public static DepartmentStatistics toDepartmentStatistics(final List<StatsEntry> stats) {
-        final var letters = stats.stream()
-            .filter(entry -> entry.originalMessageType() == LETTER && isNotEmpty(entry.department()))
-            .collect(groupingBy(StatsEntry::department,
-                groupingBy(StatsEntry::messageType,
-                groupingBy(StatsEntry::status, summingInt(n -> 1)))));
+    public static List<DepartmentStatistics> toDepartmentStatisticsList(final List<StatsEntry> stats) {
 
-        // "Extract" LETTER entries without a department and add them to the "Other" department
-        letters.putAll(stats.stream()
-            .filter(entry -> entry.originalMessageType() == LETTER && isEmpty(entry.department()))
-            .collect(groupingBy(entry -> OTHER,
+        final var statsWithOther = stats.stream()
+            .filter(entry -> entry.originalMessageType() == LETTER)
+            .map(entry -> {
+                if (isEmpty(entry.origin())) {
+                    return new StatsEntry(entry.messageType(), entry.originalMessageType(), entry.status(), OTHER, entry.department());
+                }
+                return entry;
+            }).map(entry -> {
+                if (isEmpty(entry.department())) {
+                    return new StatsEntry(entry.messageType(), entry.originalMessageType(), entry.status(), entry.origin(), OTHER);
+                }
+                return entry;
+            }).toList();
```

Comment:
I think this would be a good place to use ternary operators.

.map(entry -> isEmpty(entry.origin()) ? new StatsEntry(entry.messageType(), entry.originalMessageType(), entry.status(), OTHER, entry.department()) : entry)
.map(entry -> isEmpty(entry.department()) ? new StatsEntry(entry.messageType(), entry.originalMessageType(), entry.status(), entry.origin(), OTHER) : entry)
.toList();

Just a stylistic choice, so do what you want.